draft-ZIP-0227.rst

```diff
@@ -0,0 +1,184 @@
+- When the `finiteIssuance = 0`, the issuer can keep creating tokens of that type (in a transparent manner). The boolean value can be changed with any new issuance transaction, and could be set to `1` either when the issuer keys have been compromised, and hence stopping all issuances of that asset type (the boolean cannot be reversed), or whenever the issuer decides that the max supply has been reached.
+- Note that this mechanism can be used with the burning process to control and affect the supply of any custom asset.
+
+
+Asset Identifier Generation
+===========================
+
+For every new asset, there must be a new and unique identifier of the asset. Our design decision is not to have a chosen name for the asset type, but to delegate it to an off-chain mapping, as this would imply a land-grab “war”. From this asset identifier, which is a global byte string (valid across all Zcash protocols), we derive the specific type used within the notes, as defined by the specific protocol (e.g.: for now Orchard-style, using Pallas curve)
+
+The main requirement for the asset identifier is for it to be collision-free, as two different issuers should not be able to issue the same asset type. As such we need the asset identifier to be:
+
+- pseudorandom
+- deterministic
+- dependent on the issuer ID
+
+We define the asset identifier `assetID` (represented as a string) as
+
+:math:`\mathsf{assetID := GroupHash^{\mathbb{P}}}\mathsf{(issuerID || assetDesc)}`
+
+where
+
+- `assetDesc` is the asset description, which includes any information pertaining to the issuance.
+- `issuerID` is the public key of the issuer, used to verify the signature on the transaction and is defined as `ik` (issuer key or issuer verification key):
+
+
+:math:`\mathsf{ik := SpendAuthSig^{Orchard}.DerivePublic(isk)}`
+
+
+which uses the same signature as the spend authorization signature, where `isk` is the issuer secret key. The verification key is derived from `isk`. In order to differentiate the `isk` and `ik` keys from the actual spend authorization key, we derive another branch of keys from the main spending key `sk` as such:
+
+:math:`\mathsf{isk := ToScalar^{Orchard}(︀ PRF^{expand}_{sk} ([10])}`
+
+This allows the issuer to use the same wallet it usually uses to transfer assets, while keeping a disconnect from the other keys. It provides further anonymity and the ability to delegate issuance of an asset (or in the future, generate a multi-signature protocol) while the rest of the keys in the wallet safe.
+
+Then we let `issuanceInfo := (assetID, finiteSupply)`, where 
+
+- `assetID` is as defined above.
+- `finiteSupply` is a boolean variable that defines whether this issuance transaction of the specific asset type is the final issuance of tokens or not. Once `finiteSupply` is set, it cannot be unset. This allows expanding the functionality of the issuance mechanism:
+    - Provides `assetID` revocation in case of compromise (issue last issuance with 0 token value)
```

Comment:
How does zero token value work?
The amount of tokens can only increase, so how is their value reduced to zero?

Reply:
Can we add a way to rotate from one key to another?

This can go in a "future work" section, if we don't want to do it here.

Reply:
It is better explained in the new PR under "Concrete Applications", but what we mean is that in case an issuer needs to stop the supply generation of a given asset (without wanting to issue more tokens of such asset), the issue action can include a single note of value = 0 (such note will be added to the note commitment tree, but will be of not use as its value is 0 (aka a dummy note)
This does not mean that the amount of tokens or the total value of an asset is reduced in this action. Only burning mechanism enables this.

Also, added a future work section for key rotation in the proper sense.